cladecombiner/aggregator.py

```diff
@@ -0,0 +1,228 @@
+            The tree which we use to do the mapping.
+
+        sort_clades : bool
+            If False, mapping is done using the taxa as ordered in `targets`.
+            If True, `targets` are taxonomically sorted so that so that larger
+            `targets` do not override smaller ones. For example, if BA.2 and
+            BA.2.86 are both aggregation targets, sort_clades = True would handle
+            BA.2.86 first, such that JN.1 would map to BA.2.86, while BG.1 would
+            map to BA.2. If BA.2 is processed first, both will map to it.
+
+        unmapped_are_other : bool
+            If True, any taxon found in the input list which does not fall into
+            a target will be mapped to Taxon("other"). If False, any such taxa
+            will be mapped to themselves.
+        """
+        unknown_targets = [
+            taxon
+            for taxon in targets
+            if not taxonomy_scheme.is_valid_taxon(taxon)
+        ]
+        if len(unknown_targets) > 0:
+            raise ValueError(
+                "The following targets are not valid taxa according to the provided taxonomy scheme: "
+                + str(unknown_targets)
+            )
+        self.targets = [taxon for taxon in targets]
+        self.taxonomy_scheme = taxonomy_scheme
+        self.agg_other = unmapped_are_other
+        self.warn = warn
+        if sort_clades:
+            self.targets = sort_taxa(self.targets, self.taxonomy_scheme)
+
+    def _validate_inputs(self, input_taxa: Iterable[Taxon]) -> None:
+        unknown_taxa = [
+            taxon
+            for taxon in input_taxa
+            if not self.taxonomy_scheme.is_valid_taxon(taxon)
+        ]
+        if len(unknown_taxa) > 0:
+            raise ValueError(
+                "The following targets are not valid taxa according to the provided taxonomy scheme: "
+                + str(unknown_taxa)
+            )
+
+    def _check_missing(self, agg_map: dict[Taxon, Taxon]):
+        if self.warn:
+            used_targets = set(agg_map.values())
+            unused_targets = [
+                target for target in self.targets if target not in used_targets
+            ]
+            if len(unused_targets) > 0:
+                warn(
+                    f"The aggregation does not make use of the following input targets: {unused_targets}."
+                )
+
+    def aggregate(self, input_taxa: Iterable[Taxon]) -> Aggregation:
+        self._validate_inputs(input_taxa)
+        agg_map: dict[Taxon, Taxon] = {}
+        stack = set(input_taxa)
+        for target in self.targets:
+            children = self.taxonomy_scheme.descendants(target, True)
+            sub_map = {taxon: target for taxon in stack if taxon in children}
+            agg_map = agg_map | sub_map
+            stack.difference_update(set(agg_map.keys()))
+
+        if len(stack) > 0:
+            if self.agg_other:
+                cleanup = HomogenousAggregator(
```

Comment:
This part is cool!

So out of scope, but somehow I want the PhyloAggregator to turn itself into a SerialAggregator, with a "pure" phylogenetic aggregation bit, followed by one of these two cleanup bits. 

In my head, the default way for an aggregator to work would be to start with a trivial (self) taxon mapping, and then update it in the `for target in targets` loop. This would make the default behavior like what's in `self.agg_other = False`. And if you called with `agg_other = True`, that would trigger the creation of a serial aggregator somehow